### PR TITLE
Lets you play fetch with Ian! (Winter Thaw 2017)

### PR DESCRIPTION
**Pull request by ghost**
_This feature was created as part of the [Winter Thaw of 2017](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=9307) regardless of consensus or if it gets merged._

:cl: Xhuis
add: The HoP now starts with a squeaky rubber ball in their backpack. They can toss it around to play fetch with Ian.
/:cl:

Ian will also play with the rubber ball if it's left out in the open, and give holders puppy eyes.

### CODE REVIEW

**Comment by MrPerson** [comment]
I don't think the HoP should spawn with it. I think it should just be in his office in case someone else wants to play fetch and/or nobody played HoP that round.

**Comment by BeeSting12** [comment]
That's so cute!

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1396,3 +1397,49 @@
...
+				I.movement_target = src
+				forceMove(I)
+				break
+		if(!I.movement_target || !thrownby)
+			I.visible_message("<span class='warning'>[I] loses interest.</span>")
+			if(loc == I)
+				forceMove(get_turf(I))
+			thrownby = null
+		else
+			for(var/i = 0, i < 7, i++)
```
Comment (lzimann) [inline code/game/objects/items/toys.dm]: `for(var/i in 1 to 7)` please(same for the other one)

**Comment by WJohn** [comment]
Runtime should also be able to play with the ball, except that instead of fetching it they play by themselves and run away with it.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -359,6 +359,10 @@
 					if(isturf(S.loc) || ishuman(S.loc))
 						movement_target = S
 						break
+				for(var/obj/item/toy/rubber_ball/B in oview(src, 4))
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: This is straight copy paste from cats

Also snowflake code in general

**Review by imtakingabreakdontatme — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 2 above)
Comment (Xhuis) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: What should I be using instead? I don't know of a better way.

Comment (coiax) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: It's crude mob code to play with a ball. Of course it's snowflake.

**Comment by coiax** [comment]
Make sure a sentient Ian and Runtime can also play with the ball.

**Comment by Xhuis** [comment]
Runtime is impure and cannot play with the ball.
I'll make sure sentient animals can, though.

**Comment by dannno** [comment]
A long time ago I was going to give nuke ops a German Shepard that would fetch the nuke disk
if I made a sprite would you consider adding that

**Comment by Xhuis** [comment]
Not in this PR, but maybe in another.

**Comment by dannno** [comment]
Nienhaus has sprites in his thread actually

**Comment by Fox-McCloud** [comment]
I think this should probably be more generalized.

Most cats, dogs, and foxes all enjoy playing with rubber/tennis/etc balls.

**Comment by Xhuis** [comment]
Like I said, Ian is the only pure animal and thus is the only one capable of playing fetch.
All jokes aside, I don't really see why it should go in this PR. The request said that Ian should be able to play with the ball, so that's what I made. I can give other animals support for it in the future, or maybe you could if you want it that badly.

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Yeah but another mob does the exact same thing, so make it a proc they share

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Honestly this should fall under wanted objects that I think @RemieRichards added but thats under hostile mob code so it'd be a pain to redo it for these guys to use it.

The best ways to code this are probably not worth the time spent for something that will never be used, so making it a generic pet plays with toy proc is probably the sanest way

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Working on this right now. Should I just be putting all the searching and "playing" code under Ian's subtype or make a list of objects in the simple_animal base mob that they would seek out and interact with?

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Don't wanna do work on this in case I make a mistake. How do you want it implemented?

**Comment by KorPhaeron** [comment]
Stale, let me know when it's done

**Comment by Xhuis** [comment]
I asked you how you wanted it written twice, but you didn't respond.

**Comment by KorPhaeron** [comment]
I told you, make it a generalized proc between pets

**Comment by KorPhaeron** [comment]
or ask someone from

@tgstation/commit-access 

who actually knows how to code

**Comment by Xhuis** [comment]
I know. I wasn't sure if I should put the specific ball interaction under Ian himself or just normal pets, if normal simple mobs should have it...

Anywho, my computer is currently being formatted, so it might be a bit before AI can work on this some more.

**Comment by Xhuis** [comment]
I* goddamn phone

**Pull request closed without merging**